equinox/nn/conv.py

```diff
@@ -221,16 +221,16 @@ class Conv2d(Conv):
 
     def __init__(
         self,
-        in_channels,
-        out_channels,
-        kernel_size,
-        stride=(1, 1),
-        padding=(0, 0),
-        dilation=(1, 1),
-        groups=1,
-        use_bias=True,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: Union[int, Sequence[int]],
+        stride: Union[int, Sequence[int]] = (1, 1),
+        padding: Union[int, Sequence[int]] = (0, 0),
```

Comment:
And I think this can be a `Sequence[Tuple[int, int]]`.

Reply:
Are you sure about this? It seems to me the type can definitely be `Tuple[int, int]` but thats already covered by `Sequence[int]` (unless maybe we want to include `Tuple[int, int]` it to give the hint that the sequence must be of length 2?). Also the the documentation on `Conv` says "All of `kernel_size`, `stride`, `padding` and `dilation` can be either an integer or a sequence of integers" so I'm not sure `Sequence[Tuple[int, int]]` is valid?

Reply:
This is used to indicate asymmetric padding.

From the docs: "`padding` can alternatively also be a sequence of 2-element tuples, each representing the padding to apply before and after each spatial dimension."

Reply:
Thanks for pointing this out– fixed here and also for transpose convolutions.